src/Utilities/ErrorHandling/Exceptions.hpp

```diff
@@ -19,6 +19,13 @@ class SpectreError : public std::runtime_error {
   explicit SpectreError(const std::string& message) : runtime_error(message) {}
 };
 
+/// \ingroup ErrorHandlingGroup
+/// Exception indicating an ERROR was triggered because of an FPE
+class SpectreFpe : public std::runtime_error {
```

Comment:
Note that developers can't rely on this being thrown in all configurations, so it can't be used to try to recover from the error.

Reply:
What's the fallback if this can't be raised?

Reply:
Depending on the setup, it may abort the program (maybe call `std::terminate` or one of those things?) without reliably catching the exception, or the FPE may be ignored and execution will continue.

Reply:
Oh that second option is super fun! Yeah A note or even a warning would probably be good here.

Reply:
The first one should hopefully be rare, since it is basically a compiler-support issue and I think everything we support supports it.  That one also applies to segfaults.

The second is a hardware issue (or maybe some really low-level software, not sure), so nothing we can do about it.  It affects ARM.  (But not segfault reporting.)

Reply:
We might be able to make some progress on macOS. From this https://stackoverflow.com/questions/69059981/how-to-trap-floating-point-exceptions-on-m1-macs (and the associated long discussion) it seems FPEs do get trapped on Aarch64 on Linux, but macOS FPEs raise SIGILL. That seems to be an issue with macOS rather than AArch64. Note that the ARM toolchain doesn't support FPE trapping with AArch64, but it seems clang & gcc do.(?)